pulp_container/app/serializers.py

```diff
@@ -799,6 +811,20 @@ def validate(self, data):
                     # Append the URL of missing Artifact to the error message
                     e.detail[0] = "%s %s" % (e.detail[0], url)
                     raise e
+
+        if "repo_version" in data:
+            version_href = data["repo_version"]
+            repo_version_artifacts = RepositoryVersion.objects.get(pk=version_href.pk).artifacts
+            files = FileContent.objects.filter(
+                digest__in=repo_version_artifacts.values("sha256")
+            ).values("_artifacts__pk", "relative_path")
+            if len(files) == 0:
+                raise serializers.ValidationError(
+                    _("No file found for the specified repository version.")
+                )
+            for file in files:
+                artifacts[str(file["_artifacts__pk"])] = file["relative_path"]
```

Comment:
Image building is still considered to be [tech-preview](https://pulpproject.org/pulp_container/docs/admin/learn/tech-preview/). Have you considered changing the current API to better fit our needs? Instead of passing a list of artifacts to the task, we can try passing only the reference to the repository version, after verifying user's permissions. The task will then unpack the repository version and run the suggested queries ^.